### PR TITLE
avoid dev dep on freezegun and python-dateutil

### PR DESCRIPTION
**Pull request by graingert**
freezegun depends on python-dateutil for date parsing, python-dateutil hasn't has a release since Jul 14, 2021, and it looks like it's simpler to patch time.time to test Retry instead

### CODE REVIEW

**Comment by graingert** [comment]
it's important that Retry uses `time.time()` specifically rather than any of the other timers that freezegun also patches, eg `time.monotonic()`, because it's supposed to be parsing http Retry-After headers based on the current wall time

File: test/test_retry.py [hunk 1]
```diff
@@ -400,7 +399,7 @@ def test_respect_retry_after_header_sleep(
     ) -> None:
         retry = Retry(respect_retry_after_header=respect_retry_after_header)
 
-        with freezegun.freeze_time("2019-06-03 11:00:00", tz_offset=0), mock.patch(
+        with mock.patch("time.time", return_value=1559559600.0), mock.patch(
```
Comment (pquentin) [inline test/test_retry.py:402]: Is there a way to start from something more readable than 1559559600? What do you think about `datetime.fromisoformat("2019-06-03 11:00:00Z").timestamp()`?

Comment (graingert) [reply test/test_retry.py]: how about constructing a datetime.datetime directly?

**Review by pquentin — APPROVED** [review]
Thanks! LGTM.

**Comment by pquentin** [comment]
Wait no the failures are real. Can you please fix the tests?

**Review by sethmlarson — APPROVED** [review]

**Review by pquentin — APPROVED** [review]

**Pull request merged**